Add tests for Source Phase Imports (#3980)
tc39/test262: INTERPRETING.md

    - **`sleep`** - a function that takes a millisecond argument and
        sleeps the execution for approximately that duration.
    - **`monotonicNow`** - a function that returns a value that conforms to [`DOMHighResTimeStamp`][] and is produced in such a way that its semantics conform to **[Monotonic Clock][]**.

>     - **`AbstractModuleSource`** - a reference to the `%AbstractModuleSource%` constructor which does not appear as a property of the global object.

Author: The constructor function should be a per realm value. It is not exposed via the realm global object, so we need hosts to expose it via test262 facilities in order to test it.

>     - **`AbstractModuleSource`** - a reference to the `%AbstractModuleSource%` constructor which does not appear as a property of the global object.

Author: Ah, nice catch! I think it should be `$262.AbstractModuleSource`.

>     - **`AbstractModuleSource`** - a reference to the `%AbstractModuleSource%` constructor which does not appear as a property of the global object.

Reviewer: Could you open an issue for this? Otherwise it'll get forgotten.

Files bearing a name ending in `.json` are intended to be interpreted as JSON.

Implementers should resolve a specifier of `<do not resolve>` with a TypeError
throw completion.

> Implementers should resolve a specifier of `<abstract module source>` with an
> instance of `%AbstractModuleSource%`.

Reviewer 2: ```suggestion
Implementers should resolve the specifier `<abstract module source>` to a module that provides a valid [Module Source](https://tc39.es/proposal-source-phase-imports/#sec-module-source-objects), such as a [WebAssembly module](https://webassembly.github.io/esm-integration/js-api/index.html#webassembly-module-record).
```

And also explicitly tag those tests with a flag, such as `source-phase-imports, source-phase-imports-source` with a comment that only engines that can provide at least one module type that has a source can run `source-phase-imports-source`.

Author: Updated!

> Tests use `<module source>` specifier should be guarded with a feature flag

Reviewer 2: ```suggestion
Tests use `<module source>` specifier are guarded with a feature flag
```
Given that this is guidance for who uses the tests and not for who writes them.
